gbp/patch_series.py

```diff
@@ -117,6 +123,170 @@ def _get_subject_from_filename(self):
                 pass  # No ext so keep subject as is
         return subject.lstrip('0123456789-') or subject
 
+    def _get_info_field(self, key, get_val=None):
+        """
+        Return the key I{key} from the info C{dict}
+        or use val if I{key} is not a valid key.
+
+        Fill self.info if not already done.
+
+        @param key: key to fetch
+        @type key: C{str}
+        @param get_val: alternate value if key is not in info dict
+        @type get_val: C{()->str}
+        """
+        if self.info is None:
+            self._read_info()
+
+        if key in self.info:
+            return self.info[key]
+        else:
+            return get_val() if get_val else None
+
+    @property
+    def subject(self):
+        """
+        The patch's subject, either from the patch header or from the filename.
+        """
+        return self._get_info_field('subject', self._get_subject_from_filename)
+
+    @property
+    def author(self):
+        """The patch's author"""
+        return self._get_info_field('author')
+
+    @property
+    def email(self):
+        """The patch author's email address"""
+        return self._get_info_field('email')
+
+    @property
+    def date(self):
+        """The patch's modification time"""
+        return self._get_info_field('date')
+
```

Comment:
def email, def date move up here they need to be removed below. Othewise we end up with the identical methods on Patch and Dep3Patch

Reply:
`def author` and `def subject` too? This appears to have been a merge error, which I missed as it doesn't break any tests; good spot.